egs/wsj/s5/steps/nnet3/chain/train.py

```diff
@@ -496,17 +501,28 @@ def train(args, run_opts):
         num_archives_processed = num_archives_processed + current_num_jobs
 
     if args.stage <= num_iters:
-        logger.info("Doing final combination to produce final.mdl")
-        chain_lib.combine_models(
-            dir=args.dir, num_iters=num_iters,
-            models_to_combine=models_to_combine,
-            num_chunk_per_minibatch_str=args.num_chunk_per_minibatch,
-            egs_dir=egs_dir,
-            leaky_hmm_coefficient=args.leaky_hmm_coefficient,
-            l2_regularize=args.l2_regularize,
-            xent_regularize=args.xent_regularize,
-            run_opts=run_opts,
-            sum_to_one_penalty=args.combine_sum_to_one_penalty)
+        if args.do_final_combination:
+            logger.info("Doing final combination to produce final.mdl")
+            chain_lib.combine_models(
+                dir=args.dir, num_iters=num_iters,
+                models_to_combine=models_to_combine,
+                num_chunk_per_minibatch_str=args.num_chunk_per_minibatch,
+                egs_dir=egs_dir,
+                leaky_hmm_coefficient=args.leaky_hmm_coefficient,
+                l2_regularize=args.l2_regularize,
+                xent_regularize=args.xent_regularize,
+                run_opts=run_opts,
+                sum_to_one_penalty=args.combine_sum_to_one_penalty)
+        else:
+            logger.info("Copying the last-numbered model to final.mdl")
+            common_lib.force_symlink("{0}.mdl".format(num_iters),
+                                     "{0}/final.mdl".format(args.dir))
+            chain_lib.compute_train_cv_probabilities(
```

Comment:
Instead of calling compute_train_cv_probabilities, you could just force-symlink the final-numbered logs to the 'final' logs.